rooms.js

```diff
@@ -1242,6 +1247,20 @@ class ChatRoom extends BasicRoom {
 		this.chatRoomData = (options.isPersonal ? null : options);
 		Object.assign(this, options);
 		if (this.auth) Object.setPrototypeOf(this.auth, null);
+		/** @type {Room?} */
+		this.parent = null;
+		if (this.parent) {
```

Comment:
Wait actually very no, are you storing the parent ID in `chatRoomData`? You should definitely not give it the same name as a room object. Name it `parentid`.

Reply:
I don't know if this is what you were asking about when you said "do we NEED this right now" but if you were, I do need this before merge; literally the only thing I'm not willing to leave until after merge (besides literal crashes) is data storage format, and this is data storage (so merging means supporting it forever).

Reply:
it wasn't, I actually changed this.